app/auth/routes.py

```diff
@@ -83,3 +83,20 @@ def reset_password(token):
         flash('Your password has been reset.')
         return redirect(url_for('auth.login'))
     return render_template('auth/reset_password.html', title='Reset Password', form=form)
+
+@bp.route('/reset_password_preferences', methods=['GET', 'POST'])
+def reset_password_preferences():
+    user = current_user
+    if not user:
+        return redirect(url_for('main.index'))
+    form = ResetPasswordFormPreferences()
+    if form.validate_on_submit():
+        if (user.check_password(form.oldPassword.data)):
+            user.set_password(form.password.data)
+            db.session.commit()
+            flash('Your password has been reset.')
+            return redirect(url_for('main.index'))
```

Comment:
I thought it was a little weird being redirected to the dashboard after changing the password. 

I feel like I should of been redirect back to the settings page. 

Just a suggestion. It's your call.

Reply:
Can you also add an account history item here before the db.session.commit() please. 

You can just follow this general structure (I copied this from the user_preferences() route):
![image](https://user-images.githubusercontent.com/62105185/202608899-818bcdf6-dc44-4211-b3d7-2ebb5c6d09cf.png)

But for the description, you can just hardcode a line saying that the users password was changed.